File: go/bob/bob.go
package bob

import (
	"strings"
	"unicode"
)

func Hey(remark string) string {
	//check if remark is empty
	if strings.TrimSpace(remark) == "" {
		return 	"Fine. Be that way!"
	}
	//make the remark upper
	var testUpper = strings.ToUpper(remark)
	//test whether that changed anything
	if testUpper == remark {
		var yelling = false;
		for _, r := range remark {
			if unicode.IsLetter(r) {
				yelling = true;
			}
		}
		if yelling {
			//are we yelling a question
			if strings.Contains(remark, "?") {
				return "Calm down, I know what I'm doing!"
			} else {
				return "Whoa, chill out!"
			}
		}
	}

	//no yelling just a question ?
	if strings.HasSuffix(strings.TrimSpace(remark), "?") {
		return "Sure."
	}

	return "Whatever."
}

File: kotlin/bob/src/main/kotlin/Bob.kt
/**
 * Solution inspired by Testare's and hartror's solution.
 */
object Bob {
    fun hey(input: String): String {
        val trim = input.trim()

        if (trim.isEmpty()) return "Fine. Be that way!"

        return when(Pair(isQuestion(trim), isYelling(trim))) {
            Pair(true, false) -> "Sure."
            Pair(true, true) -> "Calm down, I know what I'm doing!"
            Pair(false, true) -> "Whoa, chill out!"
            else -> "Whatever."
        }
    }

    private fun isYelling(message : String) : Boolean {
        return message.any { it.isLetter() } && message == message.toUpperCase()
    }

    private fun isQuestion(message: String) : Boolean{
        return message.endsWith("?")
    }
}
